File: Test/AdventureWorksFunctionalModel/Sales/Currency.cs







using System;
using NakedFunctions;

namespace AW.Types {
    [Bounded]
    public class Currency {
        [Hidden]
        public string CurrencyCode { get; init; } = "";

        [Hidden]
        public string Name { get; init; } = "";

        [Hidden]
        [Versioned]
        public DateTime ModifiedDate { get; init; }

        
        public override string ToString() => $"{CurrencyCode} - {Name}";


    }
}

File: Test/NOF2.Demo.Model/Sales/Currency.vb
Namespace AW.Types


	Partial Public Class Currency
		Implements ITitledObject, INotEditableOncePersistent, IBounded

		'<Hidden>
		Public Property CurrencyCode() As String = ""

		'<Hidden>
		Public Property Name() As String = ""

		'<Hidden>
		Public Property ModifiedDate() As Date

		Public Function Title() As Title Implements ITitledObject.Title
			Return New Title(ToString())
		End Function

		Public Overrides Function ToString() As String
			Return $"{CurrencyCode} - {Name}"
		End Function
	End Class
End Namespace